content/concepts/content-routing.md

```diff
@@ -3,6 +3,112 @@ title: Content Routing
 weight: 5
 ---
 
-This article is coming soon!
+Peer-to-peer networks are inherently tricky to route because of the following 
+interdisciplinary routing challenges:
 
-Please [refer to this issue](https://github.com/libp2p/docs/issues/23) to track the progress and make suggestions.
+- The lack of universal orchestration that a central server can provide when 
+  querying and retrieving content.
+- Not having a central directory that contains information about reaching every peer 
+  in the network.
+- The presence of high node churn.
+- Data classification is communication-expensive.
+- Creating a resilient, scalable, and optimal routing protocol that is resistant to 
+  Sybil attacks and node churn while being future-proof.
+
+Content requests are made in content addressing through the CID.
+If we want to retrieve a file, we send a general request to the network with the CID.
+
+In libp2p, content routing is used for advertising content-addressed chunks of data.
+There exists a content routing interface with operations that allows 
+for content lookups.
+
+Peer routing schemes may be required to find the peers on the network based on the content 
+routing protocol; DHT-based routing is an example of this, where a DHT is used as a namespace 
+that contains peer information about how to contact different peers on the network. 
+See the peer routing guide for more details.
+
+<!-- add when published -->
+
+{{% notice "note" %}}
+While there are different design approaches for a content routing protocol, such as
+Kademlia DHT, DNS, and BitTorrent trackers, the libp2p 
+documentation will focus on a DHT-based approach that implements the content routing 
+interface: KadDHT-libp2p. Libp2p uses an adapation of the 
+[Kademila DHT algorithm](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) 
+to route peers.
+{{% /notice %}}
+
+The DHT provides a key-value store that multiple peers maintain on the network. 
+A different peer stores each row. The network knows which peer is responsible 
+for which row by a distance metric: K-closest neighbor. 
+Learn more about DHT and Kademila on the 
+[IPFS documentation](https://docs.ipfs.tech/concepts/dht/). 
```

Comment:
I would leave out the discussion of how a DHT works and just link to the DHT docs, like 

```suggestion
For more information about DHTs and the Kademlia implementation, see the [IPFS documentation](https://docs.ipfs.tech/concepts/dht/). 
```